indexes/utxoindex/src/core/api/mod.rs

```diff
@@ -47,4 +54,4 @@ pub trait UtxoIndexApi: Send + Sync {
 // 2) there is no need for an inner Arc since we hold an Arc on the Option,
 // but alas, we need Sized for the option, hence it is in a Box.
 
-pub type DynUtxoIndexApi = Arc<Option<Box<RwLock<dyn UtxoIndexApi>>>>;
+pub type DynUtxoIndexApi = Option<Box<Arc<RwLock<dyn UtxoIndexApi>>>>;
```

Comment:
I think the box is redundant in this case, since the arc has sized. I am also unsure if cloning a box of an arc increases the arc reference count.